### PR TITLE
Allow boto3 to discover credentials, add `aws_redshift_copy_role_arn` 

### PR DESCRIPTION
**Pull request by Limess**
This change allows boto3 to search for credentials automatically if an access and secret key are not provided.

This allows the target to use environment, EC2, ECS or credentials from an AWS_PROFILE file if present, rather than forcing them to be passed explicitly. Credentials will still be used by default if provided, but they are no longer required.

Added an option to explictly pass credentials for use in the `COPY` operation, this allows a different role arn to be used than that fetched from the credentials chain.

This requires a role to be setup for the integration tests which can be used by `COPY`, set to `TARGET_REDSHIFT_AWS_REDSHIFT_COPY_ROLE_ARN`.

Tested getting credentials locally using an `~/.aws/credentials` file and `AWS_PROFILE` environment variable, so other mechanisms should also work.

N.B this change means that the target will fail when trying to perform S3 operations for the first time if credentials are missing, rather than on instantiation. If failing fast is preferred I believe we could validate whether keys exist in the frozen credentials object.

Fixes https://github.com/transferwise/pipelinewise-target-redshift/issues/10

### CODE REVIEW

File: target_redshift/db_sync.py [hunk 1]
```diff
@@ -374,27 +378,36 @@ def load_csv(self, s3_key, count):
                 cur.execute(self.drop_table_query(is_stage=True))
                 cur.execute(self.create_table_query(is_stage=True))
 
-                # Step 2: Generate copy options - Override defaults from config.json if defined
+                # Step 2: Generate copy credentials - prefer role if provided, otherwise use access and secret keys
+                copy_credentials = """
+                    iam_role '{aws_role_arn}'
+                """.format(aws_role_arn=self.connection_config['aws_redshift_copy_role_arn']) if "aws_redshift_copy_role_arn" in self.connection_config else """
+                    ACCESS_KEY_ID '{aws_access_key_id}'
```
Comment (koszti) [inline target_redshift/db_sync.py]: Seems like there is a little bug here, and some cases it's causing redshift to hang on the COPY command, locking tables and not responding for long minutes.

The reason is that in `utils.py` you set `config['aws_redshift_copy_role_arn'] = None` and that producing this COPY:
```
COPY .... .csv

                    iam_role 'None'

                    EMPTYASNULL BLANKSASNULL TRIMBLANKS TRUNCATECOLUMNS
                    TIMEFORMAT 'auto'
                    COMPUPDATE OFF STATUPDATE OFF

                    DELIMITER ',' REMOVEQUOTES ESCAPE
```

Please note the `iam_role 'None'`! This puts a lock onto the table and COPY never finishes for some reason.

Replacing `if "aws_redshift_copy_role_arn" in self.connection_config` to if `self.connection_config.get("aws_redshift_copy_role_arn")` solves the problem. Can you please change it and use .get avoiding passing None?

File: tests/integration/test_target_redshift.py [hunk 2]
```diff
@@ -491,6 +491,30 @@ def test_custom_copy_options(self):
...
 
+    def test_copy_using_aws_environment(self):
+        """Test loading data with aws in the environment rather than explicitly provided access keys"""
+        tap_lines = test_utils.get_test_tap_lines('messages-with-three-streams.json')
+
+        try:
+            os.environ["AWS_ACCESS_KEY_ID"] = os.environ.get('TARGET_REDSHIFT_AWS_ACCESS_KEY')
+            os.environ["AWS_SECRET_ACCESS_KEY"] = os.environ.get('TARGET_REDSHIFT_AWS_SECRET_ACCESS_KEY')
+            config['aws_access_key_id'] = None
+            config['aws_secret_access_key'] = None
```
Comment (koszti) [inline tests/integration/test_target_redshift.py]: this needs to be `self.config['aws_access_key_id']` and `self.config['aws_secret_access_key']`
Can you please change it?

**Comment by koszti** [comment]
this is awesome. I added some comments and requested some small changes. It caused my redshift cluster to start hanging when `aws_redshift_copy_role_arn` was not set but seems like the problem found and easy to fix.

File: target_redshift/db_sync.py (hunk 1 above)
Comment (Limess) [reply target_redshift/db_sync.py]: Done, thanks for the suggestion.

File: tests/integration/test_target_redshift.py (hunk 2 above)
Comment (Limess) [reply tests/integration/test_target_redshift.py]: Done.

**Pull request merged**